src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts

```diff
@@ -239,6 +306,38 @@ export class TreeViewServerSideDemo {
     }
   };
 
+  // Non Root Server Side
+  nonRootServerSideTreeNodeData: NonRootTreeNode = NonRootTreeNode.instance;
+  nonRootServerSideDataRetriever: PagedTreeNodeDataRetriever<FoodNode> = {
+    hasChildNodes: (node: TreeNodeData<FoodNode>) => Promise.resolve(!!node.data.children && node.data.children.length > 0),
+    getChildNodeData: (node: TreeNodeData<FoodNode>) => Promise.resolve([]),
+    getPagedChildNodeData: (node: TreeNodeData<FoodNode>, pagingParams: PagingParams) => {
+      const isEmptyRoot = node instanceof NonRootTreeNode;
```

Comment:
Why creating a variable if you're only using it once ? There is no need, just add you condition to the if().

Reply:
Fixed!